src/helperFunctions/yara_binary_search.py

```diff
@@ -51,16 +62,21 @@ def _eliminate_duplicates(result_dict):
         for key in result_dict:
             result_dict[key] = sorted(set(result_dict[key]))
 
-    def get_binary_search_result(self, yara_rules):
+    def get_binary_search_result(self, task: Tuple[bytes, Optional[str]]):
         '''
-        :param yara_rules: byte string with the contents of the yara rule file
+        :param task: tuple containing the yara_rules (byte string with the contents of the yara rule file) and optionally a firmware uid if only the contents
+                     of a single firmware are to be scanned
         :return: dict of matching rules with lists of (unique) matched UIDs as values
         '''
         with NamedTemporaryFile() as temp_rule_file:
+            yara_rules, firmware_uid = task
             temp_rule_file.write(yara_rules)
             temp_rule_file.flush()
             try:
-                raw_result = self._execute_yara_search(temp_rule_file.name)
+                if firmware_uid is None:
+                    raw_result = self._execute_yara_search(temp_rule_file.name)
+                else:
+                    raw_result = self._execute_yara_search_for_single_firmware(temp_rule_file.name, firmware_uid)
             except YaraRuleError as e:
                 return e
```

Comment:
Not your code, but why **return** exception here. Would prefer returning a string if that's the whole purpose.